src/components/SaveAsTemplateModal.tsx

```diff
@@ -0,0 +1,115 @@
+import ReactModal from "react-modal";
+import { FormattedMessage } from "react-intl";
+import { useCreatePrivateTemplate } from "../queries/template-queries";
+import {
+  CampaignContent,
+  TemplateContent,
+} from "../abstractions/domain/content";
+
+interface SaveAsTemplateModalProps {
+  onClose: () => void;
+  content: TemplateContent;
+}
+
+export const SaveAsTemplateModal = ({
+  onClose,
+  content,
+}: SaveAsTemplateModalProps) => {
+  const {
+    mutate: createPrivateTemplate,
+    isLoading,
+    isSuccess,
+  } = useCreatePrivateTemplate();
+
+  const createTemplate = () => {
+    createPrivateTemplate(content);
+  };
+
+  const successContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_has_been_saved"></FormattedMessage>
+      </p>
+      <div className="form-request">
+        <div className="container-buttons">
+          <button
+            type="button"
+            name="ok"
+            className="dp-button button-medium primary-green"
+            onClick={onClose}
+          >
+            <FormattedMessage id="accept"></FormattedMessage>
+          </button>
+        </div>
+      </div>
+    </>
+  );
+
+  const formContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_description"></FormattedMessage>
+      </p>
+      <form action="#" className="form-request">
+        <fieldset>
+          <legend>Guardar como Plantilla privada</legend>
+          <ul className="field-group">
+            <li className="field-item">
+              <label htmlFor="name">
+                <FormattedMessage id="new_template_label"></FormattedMessage>
+              </label>
+              <input
+                name="name"
+                id="name"
+                defaultValue={content.templateName}
```

Comment:
How is this name being read to be used in the `createPrivateTemplate` mutation?

Reply:
I don´t understand you, but maybe now with the refactor you can read better this code part.

Reply:
I'll review it tomorrow, thanks!

Reply:
It's ok this? I did not understand the question.

Reply:
> It's ok this? 

Yes, now it will work fine.

>  I did not understand the question.

I was asking because if I was not wrong, in the previous code `createPrivateTemplate` mutation would always save the original name without reading the value from the input.

Now it is clear that the input value is being read.